### ISSUE TITLE
[Python] Implement new Concept Exercise: range

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#2176]
This issue describes how to implement the `range` type concept exercise for the Python track: should explain what `range` is and how to use it in python.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Goal of the exercises is to teach the student the use of the `range` type, an immutable sequence of numbers in python. `range` is commonly used for looping a specific number of times in `for` loops.

## Learning objectives

- understand when it is appropriate to use the `range` type
- use `range` to loop a number of times
- use `range` to loop a number of times with a specific step with `range(start, stop, step)` 
- use `range` to create a list of integer

## Out of scope

- nothing

## Concepts

- the `range` type

## Prerequisites

- `basics`
- `iterable`
- iteration
- `for` loop

## Resources to refer to

- [Python range (Python language reference)](https://docs.python.org/3/library/stdtypes.html#typesseq-range)
- [Python range on geeksforgeeks](https://www.geeksforgeeks.org/python-range-function/)
- [Python range on pynative](https://pynative.com/python-range-function/)

### Hints

- Referring to one or more of the resources linked above, or analogous resources from a trusted source.

### After

Go deeper inside the `range` call: exception handling for step passed 0 (If a step is zero Python raises a `ValueError` exception), explain how to use it with negative arguments.
Spend some words on the advantage of the `range` type over a regular `list` or `tuple`: a `range` object will always take the same (small) amount of memory, no matter the size of the range it represents (as it only stores the start, stop and step values, calculating individual items and subranges as needed).

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named comparisons_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
-

**Comment by BethanyG** [comment]
@DavidGerva @cmccandless -- I really really hate doing this.  But when I look at this concept exercise, it feels as though it belongs either with `iterables`, `iteration` or `loops` as a concept -- and not on its own as a stand-alone.  Thoughts?

**Comment by cmccandless** [comment]
I don't really think `range` is a separate concept. It's not a keyword or a type, it's just a builtin function (albeit a common one).

**Comment by BethanyG** [comment]
Yeah.  It is a "shortcut" sequence type for counting or iterating.  So that us lazy pythonistas don't have to always declare a loop counter.  ;-).  And the [python docs](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range) sorta treat it as such.  Ok.  Booting.  This concept thing is .... frustrating.  Apologies to the both of you for the churn and chaos.

**Comment by cmccandless** [comment]
> This concept thing is .... frustrating. Apologies to the both of you for the churn and chaos.

Not at all! The only stupid questions are the ones left un-asked :smile:
